python/GafferUI/MultiLineTextWidget.py

```diff
@@ -319,9 +328,57 @@ def __drop( self, widget, event ) :
 class _PlainTextEdit( QtGui.QPlainTextEdit ) :
 
 	def __init__( self, parent = None ) :
-
 		QtGui.QPlainTextEdit.__init__( self, parent )
-
+		self.__fixedLines = None
+		self.__widgetFullyBuilt = False
+		
+	def setFixedLines( self, fixedLines ) :
+		
+		self.__fixedLines = fixedLines
+		
+		self.__computeHeight()
+
+	def getFixedLines( self ) :
+
+		return self.__fixedLines
+
+	def __computeHeight( self ) :
+		
+		# since the height can be based on the font metrics when displaying a fixed number of lines, it can be only
+		# computed properly when the stylesheet has been applied to the widget which happens when the widget 
+		# is about to be shown otherwise there is no need to compute it before that (since that font metrics
+		# would be wrong anyway)
+		if not self.__widgetFullyBuilt :
+			return
+		
+		fixedLines = self.getFixedLines()
+		
+		# when the multiline displays fixed lines
+		if fixedLines is not None :
+
+			# computing the font metrics based on the number of lines
+			height = self.fontMetrics().boundingRect( "M" ).height() * fixedLines
+			
+			# also, we need to compute the widget margins to frame the fixed lines nicely 
+			margin = self.contentsMargins().top() + self.contentsMargins().bottom() + self.document().documentMargin()
+			
+			height += margin
+			
+			self.setFixedHeight(height)
+			
+		# otherwise fallbacks to the default behavior which is auto-expanded height
+		else:
+			# resetting the height
+			QWIDGETSIZE_MAX = 16777215 # qt #define not exposed by PyQt or PySide
+			self.setFixedHeight(QWIDGETSIZE_MAX)
+		
+	def showEvent ( self, event ):
+
+		self.__widgetFullyBuilt = True		
+		self.__computeHeight()
```

Comment:
Every time that Qt makes me to look through the cpp implementation to see what is really happening I think I don't know anything about Qt at all!!

The ListContainer in gaffer uses as size constraint "QLayout::SetMinAndMaxSize" which is driven by the widget minimumSize/maximumSize that is why I'm using setFixedHeight instead of sizeHint (setFixedHeight is basically a shortcut to set the same value for the minimum/maximum height).

So the sizeHint information for that constraint would not affect the limits of the widget
http://doc.qt.io/qt-4.8/qlayout.html#SizeConstraint-enum

Unless :D a custom sizePolicy gets set to the widget which is what is happening in the TextWidget, both ways work fine, but I gotta say now thinking more about it, changing the sizePolicy for the widget seems more sutiable for this case, there is just one thing I noticed that is missing in the TextWidget which is when the sizePolicy changes after the widget being displayed, so if you don't call the updateGeometry after that the widget will keep stuck displaying the geometry based in the old sizePolicy (Qt weirdness). Could I add that in the end of TextWidget.setFixedCharacterWidth ?